pkg/lib/transaction_detail.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package lib
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/moov-io/bai2/pkg/util"
+)
+
+/*
+
+Account Transaction Detail
+
+The account transaction detail record contains information about current and personal deposit
+account activity. It always has a record code of 16.
+
+*/
+
+const (
+	tdParseErrorFmt    = "AccountTransaction: unable to parse %s"
+	tdValidateErrorFmt = "AccountTransaction: invalid %s"
+)
+
+// Creating Transaction Detail
+func NewTransactionDetail() *TransactionDetail {
+	return &TransactionDetail{
+		RecordCode: "16",
+	}
+}
+
+// Transaction Detail
+type TransactionDetail struct {
+	RecordCode string
+	TypeCode   string   `json:",omitempty"`
+	Amount     string   `json:",omitempty"`
+	FundsType  string   `json:",omitempty"`
+	Composite  []string `json:",omitempty"`
+}
+
+func (h *TransactionDetail) Validate() error {
+	if h.RecordCode != "16" {
+		return fmt.Errorf(fmt.Sprintf(tdValidateErrorFmt, "RecordCode"))
+	}
+	if h.TypeCode != "" && !util.ValidateTypeCode(h.TypeCode) {
+		return fmt.Errorf(fmt.Sprintf(tdValidateErrorFmt, "TypeCode"))
+	}
+	if h.Amount != "" && !util.ValidateAmount(h.Amount) {
+		return fmt.Errorf(fmt.Sprintf(tdValidateErrorFmt, "Amount"))
+	}
+	if h.FundsType != "" && !util.ValidateFundsType(h.FundsType) {
+		return fmt.Errorf(fmt.Sprintf(tdValidateErrorFmt, "FundsType"))
+	}
+
+	return nil
+}
+
+func (h *TransactionDetail) Parse(data string) (int, error) {
+
+	var line string
+	var err error
+	var size, read int
+
+	length := util.GetSize(data)
+	if length < 2 {
+		return 0, fmt.Errorf(fmt.Sprintf(tdParseErrorFmt, "record"))
+	} else {
+		line = data[:length]
+	}
+
+	// RecordCode
+	if h.RecordCode, size, err = util.ReadField(util.GetField(line, read)); err != nil {
+		return 0, fmt.Errorf(fmt.Sprintf(tdParseErrorFmt, "RecordCode"))
+	} else {
+		read += size
+	}
+
+	// TypeCode
+	if h.TypeCode, size, err = util.ReadField(util.GetField(line, read)); err != nil {
+		return 0, fmt.Errorf(fmt.Sprintf(tdParseErrorFmt, "TypeCode"))
+	} else {
+		read += size
+	}
+
+	// Amount
+	if h.Amount, size, err = util.ReadField(util.GetField(line, read)); err != nil {
+		return 0, fmt.Errorf(fmt.Sprintf(tdParseErrorFmt, "Amount"))
+	} else {
+		read += size
+	}
+
+	// FundsType
+	if h.FundsType, size, err = util.ReadField(util.GetField(line, read)); err != nil {
+		return 0, fmt.Errorf(fmt.Sprintf(tdParseErrorFmt, "FundsType"))
+	} else {
+		read += size
+	}
+
```

Comment:
This still needs to read bank and customer reference numbers

Reply:
I think that (Bank Reference Number, Customer Reference Number, Text) are incorrect fields 
Exact field is composite field, 

this is examples: (TRANSACTION DETAIL record has 7+ fields)
16,,115,450000,S,100000,200000,150000,,,/
16,115,500000,S,,200000,300000,,,LOCK BOX NO.68751/


Composite field is very interesting field 
In fact, I reviewed a lot of sample records to figure out composite field 
they mentioned 
“Z, 0, 1 or 2” indicate the availability of the entire “Amount”.
Z = Unknown (default)

Would we expect following fields if funds type is Z (or omitted)? 
this is sample records
03,5765432,,,,,/  (funds type is Z, following fields are zero)
03,0975312468,,010,500000,,,190,70000000,4,0/ (funds type is Z, following fields are four)


Unfortunately, I don't take any way that expect following fields by each funds type 
we can put any composite sub type if the problem fixed
In this point, I used dummy composite fields with string array